plugin/src/main/java/com/redhat/jenkins/plugins/ci/authentication/activemq/SSLCertificateAuthenticationMethod.java

```diff
@@ -156,7 +156,7 @@ public FormValidation doTestConnection(@QueryParameter("broker") String broker,
             if (broker != null && isValidURL(broker)) {
                 try {
                     SSLCertificateAuthenticationMethod sam = new SSLCertificateAuthenticationMethod(keystore, Secret.fromString(keypwd), truststore, Secret.fromString(trustpwd));
-                    ActiveMQSslConnectionFactory connectionFactory = sam.getConnectionFactory(broker);
+                    ActiveMQSslConnectionFactory connectionFactory = sam.getConnectionFactory(broker + "?initialReconnectDelay=2000&maxReconnectAttempts=5");
```

Comment:
- Do we need the `initialReconnectDelay` in this use case? IIUC, It will delay first attempt by 2 seconds while user is waiting and then do its job.[1] The default of 3s for `maxReconnectDelay` seems sufficient in making sure we do not try too fast in a row.

- If I am not mistaken, the url can already contain the query arguments and appending `?` would break it.

@scoheb, can you take a look?

[1] https://activemq.apache.org/failover-transport-reference

Reply:
I need to run a test without the `initialReconnectDelay` to make sure that everything works properly

The number of attempts could be reduced to 3, and I will make that change

It failed when I entered the URL together with the "?initialReconnectDelay=2000&maxReconnectAttempts=5" in the UI

Reply:
I have seen broker entries such as:

`failover:(ssl://mybroker.acme.com:61616)
`

whether that entry makes sense is another story...but wanted to show an example of what is out there.

Reply:
Thanks, Scott!

Yes, the `failover` ones. According to the docs, we are lucky here that it accepts the *query string* portion afterwards. Like: `failover:(ssl://mybroker.acme.com:61616)?foo=bar&bax=baz`.

Have you ever come across a use-case where users needed to provide some of the options on top of what the plugin is adding already? Like `ssl://mybroker.acme.com:61616?useExponentialBackOff=true` provided by user. That, when amended by `?initialReconnectDelay=2000&maxReconnectAttempts=5`, would not form a valid URL as per the repeated `?`. Thanks!

Reply:
I have not come across such a case.

Reply:
@olivergondza 
I made a few tests without the `?initialReconnectDelay=2000` part 
And also reduce that to 0 and 1000, I found that this part is essential to the success of the test as it is